casper/src/main/resources/PoS.rho

```diff
@@ -158,8 +158,8 @@ new PoS, rs(`rho:registry:insertSigned:ed25519`), uriOut in {
                   | getParameters!(*parametersCh)
                   | for (@depositResult <- depositRet; @codeHash, @phloRate, @userPk, @timestamp <- parametersCh; lastPayment <- paymentStore) {
                     match depositResult {
-                      (true, Nil) => { paymentStore!([codeHash, userPk, timestamp, phloRate * amount]) | return!(phloRate *amount) }
-                      (false, err) => { paymentStore!([codeHash, userPk, timestamp, 0]) | return!(0) }
+                      (true, Nil) => { paymentStore!([codeHash, userPk, timestamp, amount]) | return!(true) }
+                      (false, err) => { paymentStore!([codeHash, userPk, timestamp, 0]) | return!(err) }
```

Comment:
By 'convention' I meant 'a way of encoding error results', not 'the way that we're supposed to use' :smile: 

>  It also makes value extraction more difficult in the in the computeDeployPayment function.

Yeah, I haven't got to it yet. Let's park this for now, it's probbly OK.

The benefit of having the bonding deploy's signature stored on-chain would be being able to cross-check and debug better. The deploy's signature is the closest we have to a canonical deployId.

Reply:
I think that verification is performed implicitly by the signature check in the node runtime. At the point this contract is running, we've verified that `userPk` is the true identity of the deployer.

Reply:
My point is to store the signature just in case, for better debugging ability, and the ability to do a **cross-**check - so that's another check should the previous pass unjustly. Also, deploySignature is verifying more than the deployer identity - it verifies a deploy.

I'm not saying we need it in the PoS contract - but at the same time I find it strange that the PoS contract doesn't need it.

@KentShikama WDYT?

Reply:
The deployer's identity can be accessed by the userPk right?

Reply:
The question is whether it makes sense to store the bonding deploy's signature,  so that we can later easily point to 'you've singed with your blood, here'.

Reply:
@ArturGajowy The signature check in the node verifies the validity of the deploy, which includes all of the runtime parameters used above, so the signature is implicitly included. We didn't have signature verification until recently, so we couldn't remove the signature until recently. That we can remove it demonstrates progress :)